### PR TITLE
Fix experimental step failure for unpaired sample inputs

### PR DESCRIPTION
**Pull request by lvreynoso**
In prod, unpaired sample inputs are failing in experimental. This is because the code I added to get mapped reads from the srst2 output expects that the bam file it uses will be consistently named. Unfortunately, srst2 is not consistent in naming the output sorted bam file for unpaired and paired reads and as far as we can tool there is no way to make it name the bam file in a consistent manner.

This fix finds the output sorted bam file for unpaired samples, and renames it to the expected bam filename as specified in the dag.

### CODE REVIEW

File: idseq_dag/steps/run_srst2.py [hunk 1]
```diff
@@ -94,6 +95,21 @@ def get_common_params(self):
         return ['--min_coverage', min_cov, '--threads', n_threads,
                 '--output', os.path.join(self.output_dir_local, 'output'), '--log', '--gene_db', db_file_path]
 
+    def normalize_bam_file(self):
+        """Quick fix to ensure files needed are actually present"""
```
Comment (tfrcarvalho) [inline idseq_dag/steps/run_srst2.py]: Remove "Quick fix to"

File: idseq_dag/steps/run_srst2.py [hunk 2]
```diff
@@ -94,6 +95,21 @@ def get_common_params(self):
...
+        if os.path.exists(self.output_files_local()[5]):
+            return
+        # For unpaired fastq inputs, srst2 gives a different name to the sorted bam file that it outputs
+        # We rename the bam file to what we expect (as specified in the dag)
+        elif os.path.exists(f'{self.output_dir_local}/output___R1_001.ARGannot_r2.sorted.bam'):
```
Comment (tfrcarvalho) [inline idseq_dag/steps/run_srst2.py]: Define path before and move elif -> if

File: idseq_dag/steps/run_srst2.py [hunk 3]
```diff
@@ -94,6 +95,22 @@ def get_common_params(self):
...
 
+    def normalize_bam_file(self):
+        """Ensure files needed are actually present"""
+        if os.path.exists(self.output_files_local()[5]):
+            return
+        # For unpaired fastq inputs, srst2 gives a different name to the sorted bam file that it outputs
+        # We rename the bam file to what we expect (as specified in the dag)
+        unpaired_bam_path = f'{self.output_dir_local}/output___R1_001.ARGannot_r2.sorted.bam'
+        if os.path.exists(unpaired_bam_path):
+            move_args = [unpaired_bam_path, f'{self.output_files_local()[5]}']
```
Comment (davidrissato) [inline idseq_dag/steps/run_srst2.py]: you don't need to use f strings in this case:
```
move_args = [unpaired_bam_path, self.output_files_local()[5]]
```

File: examples/26989.json [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+        "min_cov": 0,
+        "n_threads": 16,
+        "file_type": "",
+        "file_ext": "fastq"
+      }
+    }
+  ],
+  "given_targets": {
+    "fastqs": {
+      "s3_dir":  "s3://idseq-samples-prod/samples/557/26989/fastqs/", "count_reads": 1, "max_fragments": 75000000}
```
Comment (davidrissato) [inline examples/26989.json]: Shouldn't point to prod, but we can fix it later on

**Pull request merged**